### PR TITLE
Initial work support heltour.

### PR DESCRIPTION
**Pull request by lakinwecker**
This is not ready for merging. We will wait until the season is over, and also it's not complete yet. 

But I thought I would open the pull request to facilitate discussion.

### CODE REVIEW

File: config/config.js [hunk 1]
```diff
@@ -55,6 +56,14 @@ var config = {
                 "angborxley",
                 "prune2000",
             ],
+            "heltour": {
+                "token": heltour_token,
+                // TODO: this should be pulled from somewhere else so that
+                // config doesn't require updating each time a season changes
+                "base_endpoint": "https://www.lichess4545.com/api/",
```
Comment (endrawes0) [inline config/config.js]: do you mean to refer to this line with the comment above? or just the season below it?

Comment (endrawes0) [inline config/config.js]: i feel like the endpoint has to be configurable. maybe i am wrong.

Comment (lakinwecker) [inline config/config.js]: season below. Will fix.

File: src/chesster.js [hunk 2]
```diff
@@ -8,13 +8,17 @@ var winston = require("winston");
...
 // Our stuff
 var fuzzy = require('./fuzzy_match.js');
+var heltour = require('./heltour.js');
 var league = require("./league.js");
-var spreadsheets = require('./spreadsheets.js');
+var lichess = require('./lichess.js');
+var scheduling = require('./scheduling.js');
+var spreadsheet = require('./spreadsheets.js');
 var slack = require('./slack.js');
+var spreadsheets = require('./spreadsheets.js');
```
Comment (endrawes0) [inline src/chesster.js]: we have spreadsheet and spreadsheets? dont think that was intentional.

File: src/chesster.js [hunk 3]
```diff
@@ -890,7 +898,7 @@ function validateUserResult(details, result){
 
 function processGamelink(bot, message, gamelink, options, userResult){
     //get the gamelink id if one is in the message
-    var result = spreadsheets.parseGamelink(gamelink);
+    var result = spreadsheet.parseGamelink(gamelink);
```
Comment (endrawes0) [inline src/chesster.js]: whats happening here? related to my other question about spreadsheet(s)

maybe we have 2 for a reason. or our we just fumbling two variables at the moment?

File: src/chesster.js (hunk 2 above)
Comment (lakinwecker) [inline src/chesster.js]: Fumbling variables. Will fix.

File: src/chesster.js (hunk 3 above)
Comment (lakinwecker) [inline src/chesster.js]: Same, fumbling variables, thanks.

File: src/http.js [hunk 4]
```diff
@@ -1,13 +1,37 @@
 //------------------------------------------------------------------------------
```
Comment (endrawes0) [inline src/http.js]: interesting - is this more like the python http api that you were talking about than the one we require? i like it.

File: src/http.js [hunk 5]
```diff
@@ -1,13 +1,37 @@
...
+            http = _https;
+        }
+    } else {
+        if (_.isEqual(options.protocol, "https:")) {
+            http = _https;
+        }
+    }
+
+    if (options.params) {
+        options.path += "?" + querystring.stringify(options.params);
```
Comment (endrawes0) [inline src/http.js]: querystring? thats too easy. there is a library for everything

File: src/scheduling.js [hunk 6]
```diff
@@ -0,0 +1,344 @@
+var moment = require("moment");
```
Comment (endrawes0) [inline src/scheduling.js]: im assuming this stuff didnt change much. safe assumption?

File: src/http.js (hunk 4 above)
Comment (lakinwecker) [inline src/http.js]: Yes, it's moving that direction. Specifically the ability to pass in params vs bodyParameters comes from there, although they name them differently. And just writing them down now, I realize there is a difference in the spelling of "parameters" between the two. I think I will unify those.

File: test/test_scheduling.js [hunk 7]
```diff
@@ -0,0 +1,580 @@
+var assert = require('chai').assert;
```
Comment (endrawes0) [inline test/test_scheduling.js]: same for this. no changes, just moved?

**Comment by endrawes0** [comment]
Direction is great. The api looks almost like we designed it for our needs ;)

Happy with how its coming along. It will simplify much.

File: test/test_scheduling.js (hunk 7 above)
Comment (lakinwecker) [inline test/test_scheduling.js]: Yes. They should not have changed at all.

File: src/scheduling.js (hunk 6 above)
Comment (lakinwecker) [inline src/scheduling.js]: no changes. Just moved stuff around.

**Comment by lakinwecker** [comment]
Thanks for the review. I've addressed everything you mentioned.  I agree, having an API designed for our needs will simplify a lot of things.  I'll continue on with the gamelink and results next.

Then I'll do the refreshRosters / refreshPairings

**Comment by endrawes0** [comment]
I can do some. You should stop at some point and let me pick up some slack. you have been working hard.

**Pull request merged**